src/audio/igo_nr/igo_nr.c

```diff
@@ -334,7 +334,10 @@ static struct comp_dev *igo_nr_new(const struct comp_driver *drv,
 
 err:
 	rfree(dev);
-	rfree(cd);
+	if (cd) {
```

Comment:
typically the method here is to walk things out in reverse so you don't have `if` and instead have multiple labels, then adjust goto's as neccessary
e.g.
```
comp_err:
  comp_data_blob_handler_free(cd->model_handler);
err:
  rfree(cd);
  rfree(dev);
  return NULL;
```